### ISSUE TITLE
SHA256 checksum failure for jdk8u402/win32/msi

### ISSUE THREAD

**Issue by sxa** [adoptium/temurin-build#3662]
SHA256 checksum on OpenJDK8U-jre_x86-32_windows_hotspot_8u402b06.msi may not be correct.

This was flagged in a download verification test. Other files in the releases seemed ok.

**Comment by sxa** [comment]
The issue is that [this particular sha256.txt file](https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u402-b06/OpenJDK8U-jre_x86-32_windows_hotspot_8u402b06.msi.sha256.txt) contains a couple of PATH elements in the file instead of just the filename i.e.:
`bb0cb0f7d86a5b7220fceb28af5edc63e510904919e2a1e476ea24fe6ed461db  workspace/target/OpenJDK8U-jre_x86-32_windows_hotspot_8u402b06.msi`

The `workpace/target/` prefix prevents it from being usable with `sha256sum -c` - ALl other files seem correct.

**Comment by adamfarley** [comment]
The SHA256 checksum itself appears to be accurate, and the issue Stewart identified here appears to be the only problem.

File contents: bb0cb0f7d86a5b7220fceb28af5edc63e510904919e2a1e476ea24fe6ed461db  workspace/target/OpenJDK8U-jre_x86-32_windows_hotspot_8u402b06.msi

Actual msi SHA: bb0cb0f7d86a5b7220fceb28af5edc63e510904919e2a1e476ea24fe6ed461db

Stating this for posterity.

Looking into the extra path elements now.

**Comment by adamfarley** [comment]
Hmm, looks like the "workspace/target" addition is now universal, across all sha256.txt files for the latest builds of temurin.

This is not just for the jre installer now. It's for zip files and installers, jre and jdk.

Examples:

JDK8u on Windows 32bit: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x86-32-temurin/466/artifact/workspace/target/
JDK8u on Windows 64bit: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/496/artifact/workspace/target/
JDK11u on Windows 64bit: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-windows-x64-temurin/377/artifact/workspace/target/

One possibility is that this has always been the case on jenkins, and we modify the sha256 files before uploading them to the binaries repo. 

As an extension, that the issue where "only the sha256 file for the jre msi for 32bit Windows has these extra path elements" is a flaw in the binary file upload code that modifies the sha256 files prior to upload.

**Comment by adamfarley** [comment]
Ok, it looks like we normally generate the sha256.txt files during the build/pipeline (build for windows, pipeline for Linux), and that they all have extra path before the file name (inside the file). This "extra path" gets stripped off when we publish.

Based on this Slack comment [here](https://adoptium.slack.com/archives/CLCFNV2JG/p1706041905962389?thread_ts=1706024737.579909&cid=CLCFNV2JG), as well as the linked GitHub Issue comment. I'd say that these files (including the unaltered sha files for the jdk8 win32 msi files) were upload manually, skipping the step where the release script removes the paths.

TLDR: If uploading a sha file to a binaries repo manually, we need to strip out the path before the file name (inside the file).

@smlambert - Can you review this please, and say if this sounds right to you?
